### ISSUE TITLE
Mount not working but logs seem OK

### ISSUE THREAD

**Issue by gouletjg** [andyholmes/gnome-shell-extension-gsconnect#706]
**Describe the bug**

Clicking mount doesn't mount the device folders.

**Steps To Reproduce:**

1. Click mount

**Expected behavior**

Configured device folders mount in nautilus

**Support Log**

Please generate a support log ([Instructions](wiki/Help#generate-support-log)) and paste any messages related to this issue between the two ``` lines below.

```
Nov 10 14:33:14 org.gnome.Shell.Extensions.GSConnect[11186]: [/service/protocol/core.js:send:321]: Lgv20: {
                                                               "id": 1573414394580,
                                                               "type": "kdeconnect.sftp.request",
                                                               "body": {
                                                                 "startBrowsing": true
                                                               }
                                                             }
Nov 10 14:33:14 org.gnome.Shell.Extensions.GSConnect[11186]: [/service/protocol/core.js:receive/<:275]: Lgv20: {
                                                               "id": 1573414394063,
                                                               "type": "kdeconnect.sftp",
                                                               "body": {
                                                                 "ip": "x.x.x.x",
                                                                 "port": 1740,
                                                                 "user": "kdeconnect",
                                                                 "password": "XXXXXXXXXXXXXXXXX",
                                                                 "path": "/",
                                                                 "multiPaths": [
                                                                   "/SD Card",
                                                                   "/Internal"
                                                                 ],
                                                                 "pathNames": [
                                                                   "SD Card",
                                                                   "Internal"
                                                                 ]
                                                               }
                                                             }
```

**System Details (please complete the following information):**

 - **GSConnect version:** 28
   - **Installed from:** GNOME Extensions Website
 - **GNOME/Shell version:** 3.34.1
 - **Distro/Release:** Ubuntu 19.10

**GSConnect environment (if applicable):**

 - **Paired Device(s):** LGV20
 - **KDE Connect app version:** 1.13.2
 - **Plugin(s):** SFTP

**Additional Notes:**

Mount used to work, but now it doesn't.  I don't know when it stopped working as I don't use that functionality often.  I just noticed it a few days ago.  I think it may somehow be related to when I reset all settings trying to fix a minor issue.  Unrelated to this, but ever since resetting everything (as listed in the help section) I am unable to have contacts sync. I'm wondering if resetting the extension caused some issue when syncing with the app.

**Comment by andyholmes** [comment]
Sounds like a duplicate of #703. Unfortunately we never figured out the problem, it just disappeared when the reported enabled `adb` to figure out the problem.

**Comment by andyholmes** [comment]
Hmm, is this your actual IP address? `"10.11.12.204"`

**Comment by jgoulet1994** [comment]
I meant to "x" those out, but I guess I missed it.  Yes that is my IP, is it significant or are you pointing out the coincidence of"10 11 12"?

**Comment by andyholmes** [comment]
No, it just looks like a WAN IP, not a LAN IP. Some phones are known not to return the proper IP address from Android.

Are there no errors from Gvfs in your logs? This seems like there might be something going wrong in the Android app, but it's hard to say because nothing seems to be failing. You can try debugging the Android app with `adb logcat --pid=$(adb shell pidof -s org.kde.kdeconnect_tp)` and see if there are any errors on that side.

**Comment by ferdnyc** [comment]
`10.x.y.z` Class-A addresses are surprisingly common, nearly as much as `192.168`-space addresses really. They're especially popular in local networks provisioned under the control of Large™ Entities, the kind that have designs on being able to assign every client site a unique address block without any overlap. (And then they realize that there's absolutely no point in doing that with unroutable addresses, so they just quietly stop worrying about it and everyone ends up with `10.0.0.x` or `10.11.12.x` or `10.10.10.x` addresses.)

AT&T had a long-standing tradition of using the `10`-block for the non-routable private networks of their U-Verse hardware, going all the way back to the DSL days. (Not sure if they still do.)

**Comment by jgoulet1994** [comment]
Well, today I went to get the adb logcat for you and I noticed that sftp started working when I went to click the "mount" button.  The weird thing is that it was not working before I tried to get the logcat.  I don't usually connect my phone to my computer, but I think either connecting my phone via usb or starting adb actually fixed my issue.

**Comment by jgoulet1994** [comment]
I figured while I was at it, I would take a look at what is going on with why my contacts wont update.  I don't know if you want a new issue or if this is a bug in the actual app, but here are the relevant bits from the logcat.

`1-14 11:55:14.722  3765  7174 I KDE/addPlugin: Permissions OK SftpPlugin
11-14 11:55:14.725  3765  7174 I KDE/addPlugin: Optional Permissions OK SftpPlugin
11-14 11:55:14.740  3765  7174 I KDE/addPlugin: Permissions OK MprisReceiverPlugin
11-14 11:55:14.740  3765  7174 I KDE/addPlugin: Optional Permissions OK MprisReceiverPlugin
11-14 11:55:14.746  3765  7174 I KDE/addPlugin: Permissions OK BatteryPlugin
11-14 11:55:14.746  3765  7174 I KDE/addPlugin: Optional Permissions OK BatteryPlugin
11-14 11:55:14.747  3765  7174 I KDE/addPlugin: Permissions OK SharePlugin
11-14 11:55:14.749  3765  7174 I KDE/addPlugin: Optional Permissions OK SharePlugin
11-14 11:55:14.758  3765  7174 I KDE/addPlugin: Permissions OK NotificationsPlugin
11-14 11:55:14.758  3765  7174 I KDE/addPlugin: Optional Permissions OK NotificationsPlugin
11-14 11:55:14.762  3765  7174 I KDE/addPlugin: Permissions OK TelephonyPlugin
11-14 11:55:14.763  3765  7174 I KDE/addPlugin: Optional Permissions OK TelephonyPlugin
11-14 11:55:14.763  3765  7174 I KDE/addPlugin: Permissions OK PhotoPlugin
11-14 11:55:14.763  3765  7174 I KDE/addPlugin: Optional Permissions OK PhotoPlugin
11-14 11:55:14.763  3765  7174 I KDE/addPlugin: Permissions OK FindRemoteDevicePlugin
11-14 11:55:14.763  3765  7174 I KDE/addPlugin: Optional Permissions OK FindRemoteDevicePlugin
11-14 11:55:14.763  3765  7174 I KDE/addPlugin: Permissions OK ClipboardPlugin
11-14 11:55:14.763  3765  7174 I KDE/addPlugin: Optional Permissions OK ClipboardPlugin
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Permissions OK SMSPlugin
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Optional Permissions OK SMSPlugin
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Permissions OK PingPlugin
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Optional Permissions OK PingPlugin
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Permissions OK SystemVolumePlugin
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Optional Permissions OK SystemVolumePlugin
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Permissions OK FindMyPhonePlugin
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Optional Permissions OK FindMyPhonePlugin
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Permissions OK PresenterPlugin
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Optional Permissions OK PresenterPlugin
11-14 11:55:14.774  3765  7174 D RemoteKeyboardPlugin: Creating for device JohnT480sUbuntu
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Permissions OK RemoteKeyboardPlugin
11-14 11:55:14.774  3765  7174 I KDE/addPlugin: Optional Permissions OK RemoteKeyboardPlugin
11-14 11:55:14.782  3765  7174 I KDE/addPlugin: Permissions OK MprisPlugin
11-14 11:55:14.782  3765  7174 I KDE/addPlugin: Optional Permissions OK MprisPlugin
11-14 11:55:14.783  3765  7174 I KDE/addPlugin: Permissions OK ContactsPlugin
11-14 11:55:14.783  3765  7174 I KDE/addPlugin: Optional Permissions OK ContactsPlugin
11-14 11:55:14.783  3765  7174 I KDE/addPlugin: Permissions OK MousePadPlugin
11-14 11:55:14.783  3765  7174 I KDE/addPlugin: Optional Permissions OK MousePadPlugin
11-14 11:55:14.783  3765  7174 I KDE/addPlugin: Permissions OK RunCommandPlugin
11-14 11:55:14.783  3765  7174 I KDE/addPlugin: Optional Permissions OK RunCommandPlugin
11-14 11:55:14.813  3765  7175 W Device  : Ignoring packet with type kdeconnect.runcommand.request because no plugin can handle it
11-14 11:55:23.483  3765  7175 E ContactsHelper: Got an undefined type of column contact_last_updated_timestamp
11-14 11:55:23.483  3765  7175 E ContactsHelper: Got an undefined type of column contact_last_updated_timestamp
11-14 11:55:23.488  3765  7175 E KDE/Device: Exception in ContactsPlugin's onPacketReceived()
11-14 11:55:23.488  3765  7175 E KDE/Device: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
11-14 11:55:23.488  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Plugins.ContactsPlugin.ContactsPlugin.handleRequestAllUIDsTimestamps(ContactsPlugin.java:199)
11-14 11:55:23.488  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Plugins.ContactsPlugin.ContactsPlugin.onPacketReceived(ContactsPlugin.java:253)
11-14 11:55:23.488  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Device.onPacketReceived(Device.java:569)
11-14 11:55:23.488  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.BaseLink.packageReceived(BaseLink.java:84)
11-14 11:55:23.488  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.LanLink.receivedNetworkPacket(LanLink.java:255)
11-14 11:55:23.488  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.LanLink.lambda$reset$0$LanLink(LanLink.java:109)
11-14 11:55:23.488  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.-$$Lambda$LanLink$TabvaCXA5qL_bcJDmIELWusNThw.run(Unknown Source:4)
11-14 11:55:23.488  3765  7175 E KDE/Device: 	at java.lang.Thread.run(Thread.java:764)
11-14 11:55:29.301  3765  7175 E ContactsHelper: Got an undefined type of column contact_last_updated_timestamp
11-14 11:55:29.302  3765  7175 E ContactsHelper: Got an undefined type of column contact_last_updated_timestamp
11-14 11:55:29.306  3765  7175 E KDE/Device: Exception in ContactsPlugin's onPacketReceived()
11-14 11:55:29.306  3765  7175 E KDE/Device: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
11-14 11:55:29.306  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Plugins.ContactsPlugin.ContactsPlugin.handleRequestAllUIDsTimestamps(ContactsPlugin.java:199)
11-14 11:55:29.306  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Plugins.ContactsPlugin.ContactsPlugin.onPacketReceived(ContactsPlugin.java:253)
11-14 11:55:29.306  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Device.onPacketReceived(Device.java:569)
11-14 11:55:29.306  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.BaseLink.packageReceived(BaseLink.java:84)
11-14 11:55:29.306  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.LanLink.receivedNetworkPacket(LanLink.java:255)
11-14 11:55:29.306  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.LanLink.lambda$reset$0$LanLink(LanLink.java:109)
11-14 11:55:29.306  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.-$$Lambda$LanLink$TabvaCXA5qL_bcJDmIELWusNThw.run(Unknown Source:4)
11-14 11:55:29.306  3765  7175 E KDE/Device: 	at java.lang.Thread.run(Thread.java:764)
11-14 11:55:31.213  3765  7175 E ContactsHelper: Got an undefined type of column contact_last_updated_timestamp
11-14 11:55:31.213  3765  7175 E ContactsHelper: Got an undefined type of column contact_last_updated_timestamp
11-14 11:55:31.215  3765  7175 E KDE/Device: Exception in ContactsPlugin's onPacketReceived()
11-14 11:55:31.215  3765  7175 E KDE/Device: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
11-14 11:55:31.215  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Plugins.ContactsPlugin.ContactsPlugin.handleRequestAllUIDsTimestamps(ContactsPlugin.java:199)
11-14 11:55:31.215  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Plugins.ContactsPlugin.ContactsPlugin.onPacketReceived(ContactsPlugin.java:253)
11-14 11:55:31.215  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Device.onPacketReceived(Device.java:569)
11-14 11:55:31.215  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.BaseLink.packageReceived(BaseLink.java:84)
11-14 11:55:31.215  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.LanLink.receivedNetworkPacket(LanLink.java:255)
11-14 11:55:31.215  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.LanLink.lambda$reset$0$LanLink(LanLink.java:109)
11-14 11:55:31.215  3765  7175 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.-$$Lambda$LanLink$TabvaCXA5qL_bcJDmIELWusNThw.run(Unknown Source:4)
11-14 11:55:31.215  3765  7175 E KDE/Device: 	at java.lang.Thread.run(Thread.java:764)
`

**Comment by andyholmes** [comment]
> I don't usually connect my phone to my computer, but I think either connecting my phone via usb or starting adb actually fixed my issue.

That's exactly what happened to another user in #703, which is pretty frustrating.

Maybe the other error is somehow causing SFTP to fail though? You should definitely open a bug on [bugs.kde.org](https://bugs.kde.org/buglist.cgi?component=android-application&product=kdeconnect) for the Contacts helper though. Might as well ping @sredman on this contacts bug, too :wink:

**Comment by sredman** [comment]
> [kdeconnect_contacts_relevant.txt](https://github.com/andyholmes/gnome-shell-extension-gsconnect/files/3847681/kdeconnect_contacts_relevant.txt)
> 
> I figured while I was at it, I would take a look at what is going on with why my contacts wont update. I don't know if you want a new issue or if this is a bug in the actual app, but I've attached the relevant bits from the logcat. The issue seems to start in line 41 with:
> 
> `11-14 11:55:23.483 3765 7175 E ContactsHelper: Got an undefined type of column contact_last_updated_timestamp `

I don't specifically know what's going on, but the good news is by coincidence I recently rewrote the section which would have been throwing that error so whenever the next Android version is released your problem should go away

**Comment by sredman** [comment]
I don't know what's going on with SFTP, but actually I recently started having issues with it as well.... Very similar symptoms in the sense that there are no noticeable problems in the logs. That is very bizarre

**Comment by jgoulet1994** [comment]
OK, sftp stopped working again and I made two logcats. The "disabled_enabled" log begins with gsconnect disabled then I enable gsconnect, wait for a connection, and click "mount."  The other log has gsconnect already enabled and I just click "mount." 

[2019-11-16_kdeconnect_gsconnected_on.txt](https://github.com/andyholmes/gnome-shell-extension-gsconnect/files/3854616/2019-11-16_kdeconnect_gsconnected_on.txt)
[2019-11-16_kdeconnect_gsconnect_disabled_enabled.txt](https://github.com/andyholmes/gnome-shell-extension-gsconnect/files/3854617/2019-11-16_kdeconnect_gsconnect_disabled_enabled.txt)

**Comment by jgoulet1994** [comment]
I have no idea what may be useful in them.  One thing to note is that clicking "mount" did not log anything new.  The "Contacts" error from before is missing, but that may be because I disabled it on gsconnect's side.

**Comment by ferdnyc** [comment]
> One thing to note is that clicking "mount" did not log anything new.

Yeah, it's funny... I just tried that here (SFTP _is_ working for me, and hasn't had any problems)... and turns out the **same thing** is true for successful operations! I even ran with `-b all` just in case:
```sh
/adb logcat -b all --pid=$(./adb shell pidof -s org.kde.kdeconnect_tp)
```
And all the way through a Mount, browsing the remote filesystem, even opening a file, and then Files > Unmount, and this is the sum total of the log messages generated:
```
11-24 12:14:02.989   416   421 I art     : Do full code cache collection, code=247KB, data=229KB
11-24 12:14:02.990   416   421 I art     : After code cache collection, code=227KB, data=197KB
```

...Are we looking at the wrong PID or something? It wasn't my first mount operation since the devices connected, so I suppose maybe the service setup had been logged previously. Perhaps when it's just reusing the already-established service, that doesn't trigger any new output.  But Android logging is usually so noisy, it's almost creepy to not see _any_ logs over the course of a complete remote SFTP session.

**Comment by charleslimdemebrg9** [comment]
For me it worked to add the following lines in ".shh /config"

Host 'ip your phone'
KexAlgorithms diffie-hellman-group14-sha1, diffie-hellman-group1-sha1

**Comment by andyholmes** [comment]
Looks like these are being gradually [deprecated](https://tools.ietf.org/id/draft-ietf-curdle-ssh-kex-sha2-09.html#rfc.section.3.5). If this is what's been causing everyone problems, we should probably push for MINA SSHD to reconfigured or updated in kdeconnect-android.

**Comment by modzilla99** [comment]
Doesn't solve it for me at least....

**Comment by andyholmes** [comment]
Should be fixed in kdeconnect-android 1.14.2
